src/Symfony/Component/DependencyInjection/ContainerInterface.php

```diff
@@ -30,6 +30,8 @@
     /**
      * Sets a service.
      *
+     * It should be considered harmful to replace an existing service with a new instance, as it can break the dependency graph.
```

Comment:
Once the synthetic service is set, I don't think it should be allowed to be replaced. It becomes a service like any other.